daemon/daemon_unix.go

```diff
@@ -530,9 +543,51 @@ func verifyContainerResources(resources *containertypes.Resources, sysInfo *sysi
 		resources.BlkioDeviceWriteIOps = []*pblkiodev.ThrottleDevice{}
 	}
 
+	if !sysInfo.HugetlbLimit {
+		logrus.Warnf("Your kernel does not support hugetlb limit. --hugetlb-limit discarded.")
+		resources.Hugetlbs = []containertypes.Hugetlb{}
+	} else {
+		newHugetlbs, warning, err := validateHugetlbs(resources.Hugetlbs, sysInfo)
+		warnings = append(warnings, warning...)
+		if err != nil {
+			return warnings, err
+		}
+		resources.Hugetlbs = newHugetlbs
+	}
+
 	return warnings, nil
 }
 
+func validateHugetlbs(hgtlbs []containertypes.Hugetlb, sysInfo *sysinfo.SysInfo) ([]containertypes.Hugetlb, []string, error) {
+	warnings := []string{}
+	htbMap := make(map[string]uint64)
+
+	for _, hpl := range hgtlbs {
+		size, warning, err := sysInfo.ValidateHugetlb(hpl.PageSize, hpl.Limit)
+		warnings = append(warnings, warning...)
+		if err != nil {
+			return nil, warnings, err
+		}
+
+		if l, ok := htbMap[size]; ok {
+			warnings = append(warnings, fmt.Sprintf("hugetlb-limit setting of %s is repeated, former setting %d will be replaced with %d", size, l, hpl.Limit))
```

Comment:
I don't think we do this for other flags, e.g. `--label a=b --label a=c` just overwrites the previous value, so not sure we need this.